server/utils/passport/ensure-permission.js

```diff
@@ -0,0 +1,22 @@
+import UserModel from '../../models/UserModel'
+
+export default async function ensureStudyPermission(req, res, next) {
+  if (!req.isAuthenticated())
+    return res.status(401).json({ error: 'Unathorized' })
+  try {
+    const { study } = req.params
+    const { appDb } = req.app.locals
+    const user = await UserModel.findOne(appDb, { uid: req.user })
+    const { blocked, access } = user
+    const isStudyInAccess = user.access.includes(study)
+
+    if (!access.length || !isStudyInAccess)
+      return res.status(401).json({ error: 'Unathorized' })
+
+    if (blocked) return res.status(403).json({ error: 'Forbidden' })
+
+    return next()
```

Comment:
Is this middleware? Just curious about the pattern here.

Reply:
yes, this middleware. If everything here checks out then we move forward to the next set of middleware